File: Go/Easy/Class Photo/solution_1.go
package main

import "sort"

func ClassPhotos(redShirtHeights []int, blueShirtHeights []int) bool {

	sort.Ints(redShirtHeights)
	sort.Ints(blueShirtHeights)

	var firstLineColor string

	if redShirtHeights[0] > blueShirtHeights[0] {

		firstLineColor = "RED"

	} else {

		firstLineColor = "BLUE"
	}

	if firstLineColor == "RED" {

		for idx, heightOfRed := range redShirtHeights {
			heightOfBlue := blueShirtHeights[idx]

			if heightOfRed <= heightOfBlue {
				return false

			}
		}
	}

	if firstLineColor == "BLUE" {

		for idx, heightOfBlue := range blueShirtHeights {
			heightOfRed := redShirtHeights[idx]

			if heightOfBlue <= heightOfRed {
				return false

			}
		}
	}

	return true

}

File: Python/Easy/Class Photo/solution_1.py
#Sorting Algorithms takes nlog(n) time complexity. We dont store anything so its (1)
#Sum up, Time Complexity O(nlog(n)) | Space O(1)

def classPhotos(redShirtHeights, blueShirtHeights) -> bool:

    redShirtHeights.sort()
    blueShirtHeights.sort()

    shortColorFirstLine = "RED" if redShirtHeights[0] > blueShirtHeights[0] else "BLUE"
    
    if shortColorFirstLine == "BLUE":

        for idx , heightInBlue in enumerate(blueShirtHeights):
            heightInRed = redShirtHeights[idx]

            if heightInBlue <= heightInRed:
                return False

    if shortColorFirstLine == "RED":

        for idx , heightInRed in enumerate(redShirtHeights):
            heightInBlue = blueShirtHeights[idx]

            if heightInRed <= heightInBlue:
                return False
        
    return True
